src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -82,6 +82,39 @@ static void ThrowArgumentException()
             }
         }
 
+        public Guid(ReadOnlySpan<byte> b, bool bigEndian)
+        {
+            if (b.Length != 16)
+            {
+                ThrowArgumentException();
+            }
+
+            if (BitConverter.IsLittleEndian != bigEndian)
+            {
+                this = MemoryMarshal.Read<Guid>(b);
+                return;
+            }
+
+            // slower path for ReverseEndianness:
+            _k = b[15];  // hoist bounds checks
+            _a = BinaryPrimitives.ReverseEndianness(MemoryMarshal.Read<int>(b));
+            _b = BinaryPrimitives.ReverseEndianness(MemoryMarshal.Read<short>(b.Slice(4)));
+            _c = BinaryPrimitives.ReverseEndianness(MemoryMarshal.Read<short>(b.Slice(6)));
+            _d = b[8];
+            _e = b[9];
+            _f = b[10];
+            _g = b[11];
+            _h = b[12];
+            _i = b[13];
+            _j = b[14];
+
+            [StackTraceHidden]
+            static void ThrowArgumentException()
+            {
+                throw new ArgumentException(SR.Format(SR.Arg_GuidArrayCtor, "16"), nameof(b));
+            }
+        }
```

Comment:
Could we instead do something simpler like:
```C#
public Guid(ReadOnlySpan<byte> b, bool bigEndian) : this(b)
{
    if (bigEndian != BitConverter.IsLittleEndian)
    {
        _a = BinaryPrimitives.ReverseEndianness(_a);
        _b = BinaryPrimitives.ReverseEndianness(_b);
        _c = BinaryPrimitives.ReverseEndianness(_c);
    }
}
```
?

Reply:
Such code should be next:

```cs
public Guid(ReadOnlySpan<byte> b, bool bigEndian) : this(b)
{
    if (bigEndian)
    {
        _a = BinaryPrimitives.ReverseEndianness(_a);
        _b = BinaryPrimitives.ReverseEndianness(_b);
        _c = BinaryPrimitives.ReverseEndianness(_c);
    }
}
```

On BigEndian computers, it will execute the *slower path* 2 times. The `this(b)` constructor will reverse the numbers, and then the numbers will be reversed back.

Reply:
> On BigEndian computers, it will execute the slower path 2 times.

That's fine